packages/scene-composer/src/components/panels/__tests__/AddOrRemoveOverlayButton.spec.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { act, render } from '@testing-library/react';
+import wrapper from '@awsui/components-react/test-utils/dom';
+
+import { AddOrRemoveOverlayButton } from '../AddOrRemoveOverlayButton';
+import { useStore } from '../../../store';
+import { setFeatureConfig } from '../../../common/GlobalSettings';
+import { COMPOSER_FEATURES, KnownComponentType } from '../../../interfaces';
+import { Component } from '../../../models/SceneModels';
+
+jest.mock('@awsui/components-react', () => ({
+  ...jest.requireActual('@awsui/components-react'),
+}));
+
+describe('AddOrRemoveOverlayButton', () => {
+  const getSceneNodeByRef = jest.fn();
+  const addComponentInternal = jest.fn();
+  const removeComponent = jest.fn();
+
+  beforeEach(() => {
+    setFeatureConfig({ [COMPOSER_FEATURES.Overlay]: true });
+    useStore('default').setState({
+      selectedSceneNodeRef: 'selected',
+      getSceneNodeByRef,
+      addComponentInternal,
+      removeComponent,
+    });
+  });
+
+  it('should add overlay when clicking the button and the node has Tag component but no overlay component', async () => {
+    getSceneNodeByRef.mockReturnValue({ components: [{ type: KnownComponentType.Tag }] });
+
+    const { container } = render(<AddOrRemoveOverlayButton />);
+    const polarisWrapper = wrapper(container);
+
+    const addButton = polarisWrapper.findButton();
+    expect(addButton).not.toBeNull();
+
+    act(() => {
+      addButton!.click();
+    });
+
+    expect(addComponentInternal).toBeCalledTimes(1);
+    expect(addComponentInternal).toBeCalledWith(
+      'selected',
+      expect.objectContaining({
+        type: KnownComponentType.DataOverlay,
+        subType: Component.DataOverlaySubType.OverlayPanel,
+      }),
+    );
+  });
+
+  it('should render remove button when the node has Tag component and overlay component', async () => {
+    getSceneNodeByRef.mockReturnValue({
+      components: [
+        { type: KnownComponentType.Tag, ref: 'tag-ref' },
+        { type: KnownComponentType.DataOverlay, ref: 'overlay-ref' },
+      ],
+    });
+
+    const { container } = render(<AddOrRemoveOverlayButton />);
+    const polarisWrapper = wrapper(container);
+
+    const removeButton = polarisWrapper.findButton();
+    expect(removeButton).not.toBeNull();
+
+    act(() => {
+      removeButton!.click();
+    });
+
+    expect(removeComponent).toBeCalledTimes(1);
+    expect(removeComponent).toBeCalledWith('selected', 'overlay-ref');
+  });
+});
```

Comment:
I can change for this test since it's just a button and easy to find in any way, but why we need to avoid using the test utils? Have we seen issue with using those?

Reply:
Agreed, we should not rely on polaris wrapper anymore, and start removing existing references.

Reply:
can you please provide an example/scenario when using the wrapper will be problematic? just providing a statement without a reasoning or proof does not help people understand the problem and learn stuff.

Reply:
It's not testing like a user uses the application. The best-practice in front-end engineering is to test like a user, hence the popularity of `testing-library`.

Reply:
Yes, absolutely. Fundamentally, it's not the polaris wrapper itself that is problematic. It's a helper to allow you to navigate polaris's DOM. It's the implication that we are interacting and manipulating the dependent framework's internals that's the issue. It shouldn't matter to our tests how polaris structure's it's DOM at all. It's CloudScapes responsibility to ensure their components are outputting the correct styles, signature and API. From a test purity standpoint, the only part we are responsible for is validating that we call their public interface correctly (aka set their props).

This is what unit (or isolation testing) is all about. We want to isolate the thing we are testing from all outside factors, and ensure it does what it's supposed to do, nothing more. There's a principle in unit testing where each test should only have one reason to fail, and that's when the thing it's testing changes.

So it's not so much that using polarisWrapper will cause bad things to happen in our code, it's more about writing cleaner, more resilient tests that we're moving away from it. Testing how things work with their dependencies is a job for integration tests.

Reply:
I've removed the wrapper here. let's have team discussion about the standard later to get aligned.